src/components/Notification/Notification.js

```diff
@@ -229,6 +228,44 @@ export class NotificationTextDetails extends Component {
   }
 }
 
+const useIcon = kindProp =>
```

Comment:
Maybe something like:

```js
const iconTypes = {
  error: componentsX ? ErrorFilled16 : iconErrorSolid,
  success: componentsX ? CheckmarkFilled16 : iconCheckmarkSolid,
  warning: componentsX ? WarningFilled16 : iconWarningSolid,
  info: componentsX ? null : iconInfoSolid,
};
```

Then later on:

```jsx
	          <Icon
              description={iconDescription}
              className={`${prefix}--inline-notification__icon`}
              aria-label="close"
              icon={iconTypes[kind]}
            />
```

Since the info seems to be static. Is the closure for closing over `componentsX`?

Reply:
@joshblack this pattern is a holdover from https://github.com/IBM/carbon-components-react/pull/888/files#diff-005a7fa10228412b5f153e9889723cf4R158

I'm alright with refactoring but would first like some input from @asudoh on the intention behind the original pattern

Reply:
`keyValueMap[key]` evaluation here seems merely because `useIcon` is referred by multiple places and I didn't feel like doing such evaluation in multiple places. But if you feel like doing the evaluation in multiple places it'd be fine.

Reply:
@asudoh not sure what you mean by `keyValueMap[key]` evaluation. Looking at usage, it seems like calls are parameterized by a `kind` param like in: `useIcon(kind)`. The `useIcon` method itself seems to create a brand new object each time and then do the lookup. Seems like we could just statically refer to this and keep the same signature?

Reply:
@joshblack I just wanted to double-check if repeating something like `iconTypes[kind]` is what the dev wants to do - Anyhow it's just a matter of dev's taste.

Reply:
I left it as a holdover from previous commits but I am alright with refactoring as well, since makes sense to have it as just an object now